### DISCUSSION TITLE
Generating manifest.json from manifest.js

### DISCUSSION THREAD

**Discussion by maiconsanson** [amannn/next-intl#1067] [Q&A] [answered]
I'm struggling to properly generate `manifest.json` using the [metadata file](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/manifest#generate-a-manifest-file). 
Following these instructions hasn't been successful for me: https://next-intl-docs.vercel.app/docs/environments/metadata-route-handlers#metadata-files.

When I place `manifest.js` inside the `[locale]` folder, `manifest.json` isn't generated. Besides, the `{ params: { locale } }` seems inaccessible in the manifest function.

However, when I move the file outside of `[locale]`, `manifest.json` is generated but I lose access to translations.

Has anyone experienced this? Should I open an issue, or am I missing something?

## A solution that I'm avoiding
I can fix this by not using the manifest metadata file and having many static `manifest.json`.
But I have some rules in the image path to manage and a `js` file to generate the `.json` would be perfect.

```jsx
// app/[locale]/layout.js

export async function generateMetadata({ params: { locale } }) {
  return {
    manifest: `/manifest-${locale}.json`,
  }
}
```

## Repository
https://github.com/maiconsanson/next-intl-manifest
https://next-intl-manifest-git-main-maiconsansons-projects.vercel.app/

## Images
<img width="1237" alt="Screenshot 2024-05-11 at 14 29 22" src="https://github.com/amannn/next-intl/assets/1965455/e27b00cb-2623-4e04-8b62-78fcdd597f76">

<img width="635" alt="Screenshot 2024-05-11 at 11 21 19" src="https://github.com/amannn/next-intl/assets/1965455/eb5f8793-411c-41b7-85ab-48223ecb8d9b">

**Comment by amannn** [comment, marked as answer]
I've addressed this in https://github.com/amannn/next-intl/pull/1068, see [the Manifest docs](https://next-intl-docs.vercel.app/docs/environments/metadata-route-handlers#manifest).

Hope this answers your question!

**Reply by maiconsanson** [reply]
Thank you, now I understand, it works!
